Create `resultsOnly`, `segmentTrackEvent`, and `overrideSearchTerm` props for `Search` component

katydecorah opened the pull request:
This PR adds three new properties to Search:
- `resultOnly` - If true, only show results from search
- `segmentTrackEvent` - Set the segment track event, default is (`Searched docs`)
- `overrideSearchTerm` - If `resultsOnly: true` set `overrideSearchTerm` to set the search term

These properties will allow the Support team to extend the Search component to power suggestions in the new Support contact form: https://github.com/mapbox/support-center/pull/19

## How to test

1. Run `npm start`.
1. Open /Search test cases page.
1. Scroll to `Set resultsOnly to true.` test case.
1. Select an item from the test cases Select element.
1. Results for item should appear below the select.

## QA checklist

<!-- complete this checklist when adding a new component or package -->

- [x] No errors logged to console.
- [x] Accessibility score in [Chrome DevTools Audit/Lighthouse](https://developers.google.com/web/tools/lighthouse#devtools) is 100% with Lighthouse version: `#.#.#`.
- [x] Component is accessible at mobile-size viewport.
- [x] Component is accessible at tablet-size viewport.
- [x] Component is accessible at laptop-size viewport.
- [x] Component is accessible at big-monitor-size viewport.
- [ ] Create a PR in a site repo, copy the component, and test it. Push to staging and let the reviewer know they can also test the component there.

Open the test cases app locally on:

- [x] Chrome, no errors logged to console.
- [x] Firefox, no errors logged to console.
- [x] Safari, no errors logged to console.
- [x] Edge, no errors logged to console.
- [x] IE11, no errors logged to console.
- [x] Mobile Safari, no errors logged to console.
- [x] Android Chrome, no errors logged to console.

katydecorah commented on src/components/search/search.js:
```diff
@@ -78,13 +84,25 @@ class Search extends React.Component {
+  inputId: PropTypes.string,
+  /** disable modal if you always want an input */
+  disableModal: PropTypes.bool,
+  //* add current site filter toggle */
```
```suggestion
  /** add current site filter toggle */
```

adrianababakanian reviewed (commented):
These new props look great, @katydecorah ! I've added a few additional props that would also be useful for extending the Search component, which I'll push to this PR:

- `themeCompact`: a boolean that determines the text size and padding of the `SearchResult`s.
- `customEmptyResultMessage`: an optional node to us in place of the default message displayed when there are no search results for a particular query.
- I also added a loading icon to overlay the currently displayed search results when `resultsOnly: true`.

I'd appreciate your review on these changes as well. Also happy to scope this PR to the original three props that you added, and open the another PR for these changes -- let me know what you think!

katydecorah commented on src/components/search/search.js:
```diff
@@ -78,13 +86,29 @@ class Search extends React.Component {
+  /** Segment track event, default is (Searched docs) */
+  segmentTrackEvent: PropTypes.string,
+  /** If true, enable compact mode utilizing smaller text and padding, default false */
+  themeCompact: PropTypes.bool,
```
Nice!

katydecorah commented on src/components/search/search.js:
```diff
@@ -78,13 +86,29 @@ class Search extends React.Component {
+  /** If true, enable compact mode utilizing smaller text and padding, default false */
+  themeCompact: PropTypes.bool,
+  /** Node to display when there are no search results for the given query */
+  customEmptyResultMessage: PropTypes.node
```
Awesome!

katydecorah replied:
@adrianababakanian I think we could rename this prop `emptyResultMessage` and then set its defaultProps in search.js like so:

```jsx
Search.defaultProps = {
emptyResultMessage: (
    <p>
      Hmmm, we didn't find anything. Reword your
      search, or{' '}
      <a href="https://support.mapbox.com/hc/en-us">
        contact Support
      </a>
      .
    </p>
  )
}
```

Then in search-box.js, we can reference `{props.emptyResultMessage}` since the default will be set in search.js.

adrianababakanian replied:
Great suggestion @katydecorah , I like that pattern better! I'll incorporate this change.

katydecorah commented:
@adrianababakanian nice work! I left you one inline comment recommendation and pushed a few commits:

- Reverted change to package-lock.json since there are no changes to package.json.
- Ran `jest -u` to update snapshots and now the tests are passing. (I also saw that "only" test and that was an error on upstream, which I fixed and updated this PR).
- Appended a sentence to each test case description to remind us to open the Live analytics Swiftype page.

adrianababakanian reviewed (approved):
Looks great to me @katydecorah -- thanks so much for pairing on this!
